test/extended/util/framework.go

```diff
@@ -318,12 +398,27 @@ func WaitForADeployment(client kclient.ReplicationControllerInterface, name stri
 		if err != nil {
 			return err, false
 		}
+		// multiple deployments are conceivable; so we look to see how the latest depoy does
+		lastMatched := false
+		var lastErr error = nil
+		var lastRC string
 		for _, rc := range list.Items {
-			err, matched := okOrFailed(&rc)
-			if matched {
-				return err, false
```

Comment:
I had it that way for a while, especially during the time frame when we were constantly getting bit by the ICT race condition.  Once that was resolved, I reverted to this simpler change (though in hindsight now I'm not sure why I was worried about introducing a bigger change).

So sure, I can add the "find the latest RC" element to this.